src/test/prediction_service_test.cpp

```diff
@@ -1014,4 +1014,178 @@ TEST_F(TestPredict, ChangeBatchSizeViaRequestAndConfigChangeArbitraryPosition) {
     ASSERT_EQ(performInferenceWithBatchSize(response, 30, tensorflow::DataType::DT_FLOAT, batchSizePosition), ovms::StatusCode::INVALID_SHAPE);
 }
 
+/**
+ * Scenario - inference with different shapes with dynamic dummy, both dimensions reshaped to any.
+ * No model reload performed between requests.
+ *
+ * 1. Load model with input shape (-1, -1)
+ * 2. Do the inference with (3, 2) shape, expect correct output shape
+ * 3. Do the inference with (1, 4) shape, expect correct output shape
+ */
+
+TEST_F(TestPredict, PerformInferenceDummyAllDimensionsAny) {
+    using namespace ovms;
+
+    // Prepare model with changed layout to nhwc (internal layout=nchw)
+    ModelConfig config = DUMMY_MODEL_CONFIG;
+    config.setBatchingParams("0");
+    ASSERT_EQ(config.parseShapeParameter("(-1,-1)"), ovms::StatusCode::OK);
+    ASSERT_EQ(manager.reloadModelWithVersions(config), ovms::StatusCode::OK_RELOADED);
+
+    tensorflow::serving::PredictResponse response;
+
+    // Do the inference with (3, 2)
+    ASSERT_EQ(performInferenceWithShape(response, {3, 2}), ovms::StatusCode::OK);
+    checkOutputShape(response, {3, 2}, DUMMY_MODEL_OUTPUT_NAME);
+
+    // Do the inference with (1, 4)
+    ASSERT_EQ(performInferenceWithShape(response, {1, 4}), ovms::StatusCode::OK);
+    checkOutputShape(response, {1, 4}, DUMMY_MODEL_OUTPUT_NAME);
+}
+
+/**
+ * Scenario - inference with different batch sizes for dynamic batch dummy.
+ * No model reload performed between requests.
+ *
+ * 1. Load model with input shape (-1, 10)
+ * 2. Do the X inferences with (x, 10) shape, expect correct output shapes. x=(0; 100]
+ */
+
+TEST_F(TestPredict, PerformInferenceDummyBatchSizeAny) {
+    using namespace ovms;
+
+    // Prepare model with changed layout to nhwc (internal layout=nchw)
+    ModelConfig config = DUMMY_MODEL_CONFIG;
+    config.setBatchingParams("-1");
+    ASSERT_EQ(manager.reloadModelWithVersions(config), ovms::StatusCode::OK_RELOADED);
+
+    tensorflow::serving::PredictResponse response;
+
+    for (int i = 1; i <= 100; i++) {
```

Comment:
Minor:
Wouldn't it suffice to just hand-pick several values & not wast time for each inference?

Reply:
Eg. for(i : std::vector<int>({1,5,7,13,97})

Reply:
I will check the execution time of this test, I doubt this is long. We already have sequential DAG with 100 dummies which is very fast. This should be same.

Reply:
done